packages/commerce-sdk-react/src/hooks/ShopperCustomers/mutation.ts

```diff
@@ -222,6 +222,34 @@ The value of this property must be valid for the type of custom attribute define
 
 type ShopperCustomersMutationType = typeof ShopperCustomersMutations[keyof typeof ShopperCustomersMutations]
 
+const isObject = (item) => typeof item === 'object' && !Array.isArray(item) && item !== null
+
+const updateCache = (queryClient, action, queryKeysMatrix, data) => {
+    const isMatchingKey = (cacheQuery, queryKey) => {
+        return queryKey.every((item, index) => {
+            return isObject(item)
+                ? Object.values(cacheQuery.queryKey[index]).includes(item.arg)
+                : item === cacheQuery.queryKey[index]
+        })
+    }
+    // STEP 1. Update data inside query cache for the matching queryKeys
+    queryKeysMatrix[action]?.update?.map((queryKey) => {
```

Comment:
Since there is no example of a callee function in the PR, what is the mechanism here for ensuring that the `action` param when called here isn't a value that fails to match up with one of your `queryKeysMatrix` keys? This is one of those times where TypeScript would be great for type safety, but you might consider breaking your keys out as a constant that can be exported and imported from the callee context that invokes this function. e.g.

```
export const SHOPPER_ACTION_KEYS = {
     UPDATE_CUSTOMER: 'updateCustomer',
     UPDATE_CUSTOMER_ADDRESS: 'updateCustomerAddress'
}
```
and then 

```
import {SHOPPER_ACTION_KEYS} from 'hooks/ShopperCustomers/mutation'

updateCache( ..., SHOPPER_ACTION_KEYS.UPDATE_CUSTOMER_ADDRESS, ... )

```

But that might be overkill, not sure if you want or need a dictionary-type object that can be shared to sort of mimic type safety or not. Seems like it could come into play in the future, but I don't fully know the ultimate future state